EXTENSIONS.md

```diff
@@ -0,0 +1,83 @@
+# Snyk Extensions
+
+This plugin offers an extension point for integrating Snyk with other Jetbrains IDE Plugins.
+
+## What is the Snyk Controller?
+
+The Snyk Controller is, put simply, a way to control Snyk from other plugins.
+
+## Why would I use the Snyk Controller extension?
+
+There are a few of use cases for the Snyk controller:
+
+### Initiating a Snyk scan
+
+There may be situations in which a plugin wants to initiate a security scan, especially at the end of a workflow which
+introduces changes to the project source code, manifest dependencies, OCI container builds, infrastructure files,
+etc. -- anything Snyk can scan.
+
+### Determining whether Snyk is authenticated to a user
+
+It might be useful to know whether the user has authenticated with Snyk; is Snyk ready to run scans?
+
+## How do I use the extension in my plugin?
+
+### Add a dependency on this plugin to your plugin's `plugin.xml` file.
+
+Release >= 2.7.0 provides the extension point.
+
+```xml
+<depends>io.snyk.snyk-intellij-plugin</depends>
```

Comment:
Maybe mention [optional dependencies](https://plugins.jetbrains.com/docs/intellij/plugin-dependencies.html#optional-plugin-dependencies).

Reply:
Can do! I need to test that case locally to convince myself it works. You'd still end up with a compile-time dependency, but at runtime, it makes sense that it should be optional (your lifecycle manager would just never get called).